### PR TITLE
Add legacy_name param to use the original attachment names

### PR DESCRIPTION
**Pull request by tcarmeli1**
## Status
- [x] Ready

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/XSUP-39389

## Description
Add legacy_name param to use the original attachment names

### CODE REVIEW

File: Packs/EmailCommunication/ReleaseNotes/2_0_30.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+#### Scripts
+
+##### PreprocessEmail
+
+- Updated the script to handle both attachments names in the original format (eg., image.png) and in the new format <Attachment-ID\>-attachmentName-<Attachment-name\> (eg., 123-attachmentName-image.png).
```
Comment (ShirleyDenkberg) [inline Packs/EmailCommunication/ReleaseNotes/2_0_30.md:6]:
```suggestion
Updated the script to handle both attachments names in the original format (e.g., image.png) and in the new format <Attachment-ID\>-attachmentName-<Attachment-name\> (e.g., 123-attachmentName-image.png).
```

File: Packs/Gmail/ReleaseNotes/1_3_22.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### Gmail
+
+- Updated the configuration parameters- added a Use Legacy Name param in order to use the plain attachment name.
```
Comment (ShirleyDenkberg) [inline Packs/Gmail/ReleaseNotes/1_3_22.md:6]:
```suggestion
- Added a Use Legacy Name configuration parameter in order to use the plain attachment name.
```

File: Packs/Gmail/ReleaseNotes/1_3_22.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+- Updated the attachment name to be in the following format : <the attachment ID\>-attachmentName-<the attachment name\> only if the attachment is inline and **Use Legacy Name** is disabled.
```
Comment (ShirleyDenkberg) [inline Packs/Gmail/ReleaseNotes/1_3_22.md:7]:
```suggestion
- Updated the attachment name to be in the following format: <the attachment ID\>-attachmentName-<the attachment name\> only if the attachment is inline and **Use Legacy Name** is disabled.
```

File: Packs/GmailSingleUser/ReleaseNotes/1_4_3.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### Gmail Single User
+
+- Updated the configuration parameters- added a Use Legacy Name param in order to use the plain attachment name.
+- Updated the attachment name to be in the following format : <the attachment ID\>-attachmentName-<the attachment name\> only if the attachment is inline and **Use Legacy Name** is disabled.
```
Comment (ShirleyDenkberg) [inline Packs/GmailSingleUser/ReleaseNotes/1_4_3.md:7]:
```suggestion
- Added a Use Legacy Name configuration parameter in order to use the plain attachment name.
- Updated the attachment name to be in the following format: <the attachment ID\>-attachmentName-<the attachment name\> only if the attachment is inline and **Use Legacy Name** is disabled.
```

File: Packs/MicrosoftExchangeOnPremise/ReleaseNotes/2_1_10.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### EWS v2
+
+- Updated the configuration parameters- added a Use Legacy Name param in order to use the plain attachment name.
+- Updated the attachment name to be in the following format : <the attachment ID\>-attachmentName-<the attachment name\> only if the attachment is inline and **Use Legacy Name** is disabled.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftExchangeOnPremise/ReleaseNotes/2_1_10.md:7]:
```suggestion
- Added a Use Legacy Name configuration parameter in order to use the plain attachment name.
- Updated the attachment name to be in the following format: <the attachment ID\>-attachmentName-<the attachment name\> only if the attachment is inline and **Use Legacy Name** is disabled.
```

File: Packs/MicrosoftExchangeOnline/ReleaseNotes/1_4_4.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### EWS O365
+
+- Updated the configuration parameters- added a Use Legacy Name param in order to use the plain attachment name.
+- Updated the attachment name to be in the following format : <the attachment ID\>-attachmentName-<the attachment name\> only if the attachment is inline and **Use Legacy Name** is disabled.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftExchangeOnline/ReleaseNotes/1_4_4.md:7]:
```suggestion
- Added a Use Legacy Name configuration parameter in order to use the plain attachment name.
- Updated the attachment name to be in the following format: <the attachment ID\>-attachmentName-<the attachment name\> only if the attachment is inline and **Use Legacy Name** is disabled.
```

File: Packs/MicrosoftGraphMail/ReleaseNotes/1_6_13.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
+
+#### Integrations
+
+##### Microsoft Graph Mail Single User
+
+- Updated the configuration parameters- added a Use Legacy Name param in order to use the plain attachment name.
+- Updated the attachment name to be in the following format : <the attachment ID\>-attachmentName-<the attachment name\> only if the attachment is inline and **Use Legacy Name** is disabled.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphMail/ReleaseNotes/1_6_13.md:7]:
```suggestion
- Added a Use Legacy Name configuration parameter in order to use the plain attachment name.
- Updated the attachment name to be in the following format: <the attachment ID\>-attachmentName-<the attachment name\> only if the attachment is inline and **Use Legacy Name** is disabled.
```

File: Packs/MicrosoftGraphMail/ReleaseNotes/1_6_13.md [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+##### O365 Outlook Mail (Using Graph API)
+
+- Updated the configuration parameters- added a Use Legacy Name param in order to use the plain attachment name.
+- Updated the attachment name to be in the following format : <the attachment ID\>-attachmentName-<the attachment name\> only if the attachment is inline and **Use Legacy Name** is disabled.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphMail/ReleaseNotes/1_6_13.md:12]:
```suggestion
- Added a Use Legacy Name configuration parameter in order to use the plain attachment name.
- Updated the attachment name to be in the following format: <the attachment ID\>-attachmentName-<the attachment name\> only if the attachment is inline and **Use Legacy Name** is disabled.
```

**Comment by ShirleyDenkberg** [comment]
@jbabazadeh Doc review completed.

**Comment by tcarmeli1** [comment]
Hi @yuvalbenshalom ,
Can you please assist with a force merge on this PR.
It fails because of the following reasons:
secret detection- https://panw-global.slack.com/archives/C064JN9C37Z/p1722242373769769 known failure.
Gmail single user TPB- has a ticket https://jira-dc.paloaltonetworks.com/browse/CRTX-122260.
Send investigation summary reports TPB- has a ticket https://jira-dc.paloaltonetworks.com/browse/CIAC-9721
Detect & Manage Phishing Campaigns - Test TPB- succeed on xsoar6 , fails on xsoar 8 before and after my changes.
BC104 in all 6 integrations- deleted the hidden argument i added in the original PR for filtering out files which where already downloaded.

**Pull request merged**